### ISSUE TITLE
SA1633 shows a code fix even if the header is correct.

### ISSUE THREAD

**Issue by MatthieuNarcisi** [DotNetAnalyzers/StyleCopAnalyzers#1627]
We use headers in our CS files with this format :

```csharp
//-----------------------------------------------------------------------
// <copyright file="ViewModel.cs" company="Company">
// Copyright (c) Company. All rights reserved.
// </copyright>
// <author>FirstName LastName</author>
//-----------------------------------------------------------------------
```

This format is valid as described here : https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1633.md

But stylecop is reporting it as a violation and suggest the following code fix : 

```csharp
// <copyright file="ViewModel.cs" company="Company">
// Copyright (c) Company. All rights reserved.
// </copyright>
```
We are using version 1.0.0-beta013
I am using

**Comment by sharwell** [comment]
@vweijsters Does the file header extraction not currently support bars?

**Comment by vweijsters** [comment]
Assign it to me, I'll have a look

**Comment by vweijsters** [comment]
There is a test case for this in `FileHeaderTestBase.TestValidFileHeaderWithBordersAsync`
The code however is very strict in formatting.

@MatthieuNarcisi Do your headers have any spaces (or other characters) between the `//` and the `-` characters?

**Comment by MatthieuNarcisi** [comment]
I think I found the issue. It is (kind of) my fault.

My default header (when I create a class) look like this :
```csharp
//-----------------------------------------------------------------------
// <copyright file="ViewModel.cs" company="Company">
//     Copyright (c) Company. All rights reserved.
// </copyright>
// <author><FullName></author>
//-----------------------------------------------------------------------
```
There is a XML tag that the user has to replace by their name.
The error is only reported when I have this default header before I change it, so my guess is that the parser tries to parse the XML but fails (as the FullName tag is not closed) and report an error.

**Comment by sharwell** [comment]
@MatthieuNarcisi Perhaps `{FullName}` would be a better alternative?

**Comment by sharwell** [comment]
:question: @MatthieuNarcisi What is the exact message reported by StyleCop.Analyzers in your case? Does it indicate that the header does not contain valid XML?

**Comment by MatthieuNarcisi** [comment]
@sharwell Indeed the exact message is :

```
Warning	SA1633	The file header XML is invalid.
```

You are right, my understanding of the error was wrong. It was simply that the code fix was misleading, I though the whole header was wrong. But I agree with you, the improvement to the project is minimal.

Thank you for your time and help anyway.

**Comment by sharwell** [comment]
@MatthieuNarcisi I filed #1633 with a more specific enhancement request that would likely improve the user experience in scenarios like yours. :+1:

**Comment by NeftalyPerez1** [comment]
this is a little be late now but you guys build the json file first which is fine also you guy need to set explicit access modifiers and XML comments for the class and method
